Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs

```diff
@@ -86,14 +85,16 @@ async void UpdatePushEnabled(object sender, ToggledEventArgs e)
 
         async void RefreshDistributeEnabled(bool _appCenterEnabled)
         {
-            DistributeEnabledSwitchCell.On = await Distribute.IsEnabledAsync();
-            DistributeEnabledSwitchCell.IsEnabled = _appCenterEnabled;
-            RefreshDistributeTrackUpdate();
+            var distributeEnabled = await Distribute.IsEnabledAsync();
+            DistributeEnabledSwitchCell.On = distributeEnabled;
+            DistributeEnabledSwitchCell.IsEnabled = _appCenterEnabled && distributeEnabled;
+            RefreshDistributeTrackUpdate(distributeEnabled);
+            RefreshAutomaticUpdateCheck(_appCenterEnabled && distributeEnabled);
         }
 
-        async void RefreshDistributeTrackUpdate()
+        async void RefreshDistributeTrackUpdate(bool? _distributeEnabled = null)
         {
-            var isDistributeEnable = await Distribute.IsEnabledAsync();
+            var isDistributeEnable = _distributeEnabled.HasValue ? _distributeEnabled.Value : await Distribute.IsEnabledAsync();
```

Comment:
Update track state does not depend on distribute enabled state, we should revert those changes.